staging/src/k8s.io/api/resource/v1beta2/types.go

```diff
@@ -833,6 +965,48 @@ type DeviceSubRequest struct {
 	// +listType=atomic
 	// +featureGate=DRADeviceTaints
 	Tolerations []DeviceToleration `json:"tolerations,omitempty" protobuf:"bytes,6,opt,name=tolerations"`
+
+	// Capacity define resource requirements against each capacity.
+	//
+	// If this field is unset and the device supports multiple allocations,
+	// the default value will be applied to each capacity with a defined sharing policy.
+	//
+	// Applies to each device allocation.
+	// If Count > 1,
+	// request fails if there aren't enough devices that meet the requirements.
+	// If AllocationMode is set to All,
+	// request fails if any device doesn't meet the requirements.
+	//
+	// +optional
+	// +featureGate=DRAConsumableCapacity
+	Capacity *CapacityRequirements `json:"capacity,omitempty" protobuf:"bytes,7,opt,name=capacity"`
+}
+
+// CapacityRequirements defines the capacity requirements for a specific device request.
+type CapacityRequirements struct {
+	// Requests represent individual device resource requests for distinct resources,
+	// all of which must be provided by the device.
+	//
+	// For each device capacity with a defined sharingPolicy,
+	// the scheduler uses the policy to determine how much capacity is consumed:
+	// - If no capacity request is specified, the device sharingPolicy.default value is used.
+	// - If the device sharingPolicy defines a validRange,
+	//   the capacity request is rounded up to the nearest valid value in the range.
+	// - If the device sharingPolicy defines validValues, the capacity request is rounded up to the nearest valid value.
+	// The consumed capacity is written to the resource claim status.devices[*].consumedCapacity field.
+	//
+	// This value is used as an additional filtering condition against the available capacity on the device.
+	// This is semantically equivalent to a CEL selector with
+	// `device.capacity[<domain>].<name>.compareTo(quantity(<request quantity>)) >= 0`.
+	// For example, device.capacity['test-driver.cdi.k8s.io'].counters.compareTo(quantity('2')) >= 0.
+	//
+	// +optional
+	Requests map[QualifiedName]resource.Quantity `json:"requests,omitempty" protobuf:"bytes,1,rep,name=requests,castkey=QualifiedName"`
+
+	// Potential extension:
+	// `Limits` field to describe burstable consumption.
+	// Handling burstability would be the responsibility of the individual device driver,
+	// similar to how the CPU manager handles CPU burst behavior.
```

Comment:
hoist these out of the PR and into a "future possibilities" section of the KEP at this point